src/lwc/geFormRenderer/geFormRenderer.js

```diff
@@ -694,25 +694,72 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
         }
     }
 
+    handleChangePicklist(event) {
+        const account = DATA_IMPORT_ACCOUNT1_IMPORTED_FIELD.fieldApiName;
+        const contact = DATA_IMPORT_CONTACT1_IMPORTED_FIELD.fieldApiName;
+        const donorTypeFieldApiName = DONATION_DONOR_FIELDS.donationDonorField;
+        const picklistFieldApiName = event.detail.fieldApiName;
+        const picklistValue = event.detail.value;
+
+        if (picklistFieldApiName === donorTypeFieldApiName) {
+            const sectionsList = this.template.querySelectorAll('c-ge-form-section');
+            const sectionData = this.getData(sectionsList);
+            const diRecord = sectionData.diRecord;
+            const picklistDonorType = picklistValue === 'Account1' ? 'Account' : 'Contact';
+            const recordId = picklistDonorType === 'Account' ? diRecord[account] : diRecord[contact];
+
+            this.setReviewDonationsDonorProperties(recordId, picklistDonorType);
+        }
+    }
+
     // TODO: Need to handle displaying of review donations onload when coming from an Account/Contact page
     handleChangeLookup(event) {
-        const detail = event.detail;
+        const recordId = event.detail.recordId;
+        const lookupFieldApiName = event.detail.fieldApiName;
         const account = DATA_IMPORT_ACCOUNT1_IMPORTED_FIELD.fieldApiName;
         const contact = DATA_IMPORT_CONTACT1_IMPORTED_FIELD.fieldApiName;
+        const lookupDonorType = lookupFieldApiName === account ? 'Account' : 'Contact';
```

Comment:
Technically only Account1 and Contact1 qualify as donors - does this update the donation matching for Contact2 and Account2 as well?